packages/core/src/processing/generator/generated_data.rs

```diff
@@ -78,11 +80,34 @@ impl GeneratedData {
         };
 
         // write header and records
-        for r in synthetic_data.iter() {
-            match wtr.write_record(r.iter().map(|v| v.as_str())) {
+        if long_form {
+            let col_headers = &synthetic_data[0];
+            let long_form_headers = ["Id", "Attribute", "Value", "AttributeValue"];
+            match wtr.write_record(long_form_headers) {
                 Ok(_) => {}
                 Err(err) => return Err(CsvIOError::new(err)),
             };
+            for (row_idx, r) in synthetic_data.iter().skip(1).enumerate() {
+                for (col_idx, c) in r.iter().enumerate() {
+                    let long_form_row = [
```

Comment:
I believe `write_record` below will replicate the memory by writing to the writer anyways. So perhaps in here we could work with string references to avoid one more memory duplication. Maybe something like:

```rust
  let long_form_row = [
      &row_idx.to_string(),
      &col_headers[col_idx],
      c,
      &format!("{}:{}", col_headers[col_idx], c),
  ];
```

This produces a `[&String; 4]` instead of `[String; 4]`.